### PR TITLE
feat(REST): Added new Rest API endpoint for reading clearing request

### PR DESCRIPTION
**Pull request by akapti**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

Created the `REST` endpoint to get the `clearing request` details based on 

1. Clearing request Id
2. Project id

* Which issue is this pull request belonging to and how is it solving it? 
#1156 
* Did you add or update any new dependencies that are required for your change? 
No

Issue: #1156 

* User should be able to see the clearing request details by performing `Get` operation at endpoint:

1.  `/api/clearingrequest/{CrId}` where `Clearing Request id` is the URL param.
2.  `/api/clearingrequest/project/{ProjectId}` where `Project Id` is the URL param.

 
> Have you implemented any additional tests?
No.

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

Signed-off-by: Abdul Kapti <abdul.kapti@siemens-healthineers.com>

### CODE REVIEW

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/clearingrequest/ClearingRequestController.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        ClearingRequest clearingRequest = sw360ClearingRequestService.getClearingRequestByProjectId(projectId, sw360User);
+
+        HttpStatus status = clearingRequest == null ? HttpStatus.NO_CONTENT : HttpStatus.OK;
+        return new ResponseEntity(clearingRequest, status);
+    }
+
+    @Override
+    public RepositoryLinksResource process(RepositoryLinksResource resource) {
+        final ControllerLinkBuilder controllerLinkBuilder = linkTo(ClearingRequestController.class);
+        final Link changelogLink = new Link(
```
Comment (smrutis1) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/clearingrequest/ClearingRequestController.java:96]: please consider to update variable name

Comment (akapti) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/clearingrequest/ClearingRequestController.java]: changed the variable name.

**Comment by smrutis1** [comment]
Tested it. Working fine.

**Review by smrutis1 — APPROVED** [review]
Code looks good

**Pull request merged**